WeakAuras/Profiling.lua

```diff
@@ -723,60 +525,273 @@ function RealTimeProfilingWindow:Init()
       self.is_moving = nil
     end
   end)
-
+]]
   self:SetScript("OnUpdate", self.RefreshBars)
-  self.init = true
   self:UpdateButtons()
 end
 
-function RealTimeProfilingWindow:UpdateButtons()
-  if not self.init then
-    return
+function WeakAurasProfilingResultButton_OnClick(self)
+  WeakAuras.PrintProfile()
+end
+
+local function nextEncounterButton_OnClick(self)
+  if currentProfileState ~= "encounter" then
+    WeakAuras.StartProfile("encounter")
   end
-  if currentProfileState == "combat" then
-    self.combatButton:SetText(L["Cancel"])
+  local parent = self:GetParent()
+  local profilingFrame = parent.dropdown.Button:GetParent():GetParent():GetParent()
+  profilingFrame:ResetBars()
+  profilingFrame:UpdateButtons()
+end
+
+local function nextCombatButton_OnClick(self)
+  if currentProfileState ~= "combat" then
+    WeakAuras.StartProfile("combat")
+  end
+  local parent = self:GetParent()
+  local profilingFrame = parent.dropdown.Button:GetParent():GetParent():GetParent()
+  profilingFrame:ResetBars()
+  profilingFrame:UpdateButtons()
+end
+
+local function startNowButton_OnClick(self)
+  WeakAuras.StartProfile()
+  local parent = self:GetParent()
+  local profilingFrame = parent.dropdown.Button:GetParent():GetParent():GetParent()
+  profilingFrame:ResetBars()
+  profilingFrame:UpdateButtons()
+end
+
+function WeakAurasProfilingStopButton_OnClick(self)
+  if currentProfileState == "profiling" then
+    WeakAuras.StopProfile()
   else
-    self.combatButton:SetText(L["Next Combat"])
+    CancelScheduledProfile()
+  end
+  self:GetParent():GetParent():UpdateButtons()
+end
+
+function WeakAurasProfilingMixin:InitDropDown()
+  local function Initializer(dropDown, level)
+    local entries = {
+      {
+        text = L["Start Now"],
+        func = startNowButton_OnClick,
+      },
+      {
+        text = L["Next Encounter"],
+        func = nextEncounterButton_OnClick,
+      },
+      {
+        text = L["Next Combat"],
+        func = nextCombatButton_OnClick,
+      },
+    }
+		for _, entry in ipairs(entries) do
+			local info = UIDropDownMenu_CreateInfo()
+			info.notCheckable = true
+			info.text = entry.text
+			info.func = entry.func
+			UIDropDownMenu_AddButton(info)
+		end
+	end
+
+	local dropDown = self.buttons.startDropDown
+  local dropDownButton = self.buttons.start
```

Comment:
Is there any way we can make the 3 widgets in the top line have the same size?

Reply:
The dropdown in the middle height is weird due to the texture it use, i think this is slightly better

![image](https://github.com/WeakAuras/WeakAuras2/assets/189656/0ec9dd4b-de26-4a96-8328-fca67245db6b)